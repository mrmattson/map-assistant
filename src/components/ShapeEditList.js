import React from 'react'
import PropTypes from 'prop-types'
import is from 'is_js'
import R from 'ramda'
import {
  Button
} from 'react-bootstrap'

import ShapeEditItem from './ShapeEditItem'

function ShapeEditList ({ draftShapes, selectedDraftShapes, saveShapesAsFinal, onSelectionChange, onPropertyChange }) {
  return (
    <div>
      <ul className="ShapeEditList list-group">
        {is.not.empty(draftShapes)
          ? draftShapes.map((draftShape, index) => (
            <ShapeEditItem
              key={draftShape.properties.id}
              draftShape={draftShape}
              selectedDraftShapes={selectedDraftShapes}
              index={index}
              onSelectionChange={onSelectionChange}
              onPropertyChange={onPropertyChange}
            />))
          : null
        }
      </ul>
      <Button
        bsSize="large"
        bsStyle="primary"
        onClick={() => saveShapesAsFinal(R.filter(draftShape => R.contains(draftShape.properties.id, selectedDraftShapes), draftShapes))}
        disabled={draftShapes.length === 0}
      >Save Selected Shapes</Button>
    </div>
  )
}

// https://facebook.github.io/react/docs/typechecking-with-proptypes.html
ShapeEditList.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  draftShapes: PropTypes.array.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  selectedDraftShapes: PropTypes.array.isRequired,
  saveShapesAsFinal: PropTypes.func.isRequired,
  onSelectionChange: PropTypes.func.isRequired,
  onPropertyChange: PropTypes.func.isRequired
}

export default ShapeEditList
