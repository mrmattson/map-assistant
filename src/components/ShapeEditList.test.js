/* eslint-env jest */

import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import ShapeEditList from './ShapeEditList'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(
    <ShapeEditList
      draftShapes={[]}
      selectedDraftShapes={[]}
      saveShapesAsFinal={function saveShapesAsFinal () { return 0 }}
      onSelectionChange={function onSelectionChange () { return 0 }}
      onPropertyChange={function onPropertyChange () { return 0 }}
    />, div)
})

it('clean snapshot', () => {
  const component = renderer.create(
    <ShapeEditList
      draftShapes={[]}
      selectedDraftShapes={[]}
      saveShapesAsFinal={function saveShapesAsFinal () { return 0 }}
      onSelectionChange={function onSelectionChange () { return 0 }}
      onPropertyChange={function onPropertyChange () { return 0 }}
    />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
