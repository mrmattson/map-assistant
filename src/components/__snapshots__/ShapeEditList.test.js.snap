exports[`test clean snapshot 1`] = `
<div>
  <ul
    className="ShapeEditList list-group" />
  <button
    className="btn btn-lg btn-primary"
    disabled={true}
    onClick={[Function]}
    type="button">
    Save Selected Shapes
  </button>
</div>
`;
